code/game/machinery/doors/airlock.dm

```diff
@@ -190,11 +189,8 @@ GLOBAL_LIST_EMPTY(airlock_emissive_underlays)
 /obj/machinery/door/airlock/bumpopen(mob/living/user) //Airlocks now zap you when you 'bump' them open when they're electrified. --NeoFite
 	if(!issilicon(usr))
 		if(isElectrified())
-			if(!justzap)
+			if(!shockCooldown <= world.time)
```

Comment:
I think your logic is a little mixed up here: the ! will apply to the shock cool down variable here

```suggestion
			if(shockCooldown <= world.time)
```

I'm pretty sure I'm getting this right, just want to bring it to your attention